### PR TITLE
updated common.abstract to indicate failing method in error message

### PR DESCRIPTION
**Pull request by sakekasi**

### CODE REVIEW

File: src/common.js [hunk 1]
```diff
@@ -30,10 +30,15 @@ escapeStringFor['\u000b'.charCodeAt(0)] = '\\v';
 // Exports
 // --------------------------------------------------------------------
 
-exports.abstract = function() {
-  throw new Error(
-      'this method is abstract! ' +
+exports.abstract = function(methodName) {
```
Comment (alexwarth) [inline src/common.js]: s/methodName/optMethodName

File: src/common.js [hunk 2]
```diff
@@ -30,10 +30,15 @@ escapeStringFor['\u000b'.charCodeAt(0)] = '\\v';
...
+  if (!methodName) {
```
Comment (alexwarth) [inline src/common.js]: var methodName = optMethodName || '';

File: src/common.js [hunk 3]
```diff
@@ -30,10 +30,15 @@ escapeStringFor['\u000b'.charCodeAt(0)] = '\\v';
...
+    methodName = '';
+  }
+  return function() {
+    throw new Error(
+      'this method(' + methodName + ') is abstract! ' +
```
Comment (alexwarth) [inline src/common.js]: How about: `'method ' + methodName + ' is abstract! ' + ...` ?

**Comment by alexwarth** [comment]
Thanks, Saketh! I just wrote a couple of comments / nits. Let me know when you address them and I'll accept.

**Comment by pdubroy** [comment]
🙌  This has been bugging me for a while. Thanks for fixing it!

File: test/test-errors.js [hunk 4]
```diff
@@ -200,3 +200,16 @@ test('trailing space should not influence the result', function(t) {
   t.equal(failures[0].type, 'string');
   t.end();
 });
+
+test('method name displayed on abstract function failure', function(t) {
+  var g = ohm.ohmGrammar;
+  var seq = g.ruleBodies.Grammar;
+  try {
```
Comment (pdubroy) [inline test/test-errors.js]: FYI, you can test this more succinctly like this:

    t.throws(function() { seq.toFailure(); }, /this method(toFailure) is abstract!/);

**Pull request merged**